src/org/thoughtcrime/securesms/BaseActionBarActivity.java

```diff
@@ -76,6 +66,12 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
 
   @Override
   protected void onResume() {
+    if (shouldLock()) {
+      ScreenLockUtil.applyScreenLock(this);
```

Comment:
Only here do we need to make sure not to apply the screen lock twice.
This must be an inner selection clause because the `else if` below does not check on `isWaitingForResult` again.

```suggestion
      if(!isWaitingForResult)
        ScreenLockUtil.applyScreenLock(this);
```

Reply:
But in `onActivityResult()` `isWaitingForResult` is always set to `false`, so that this new condition will always be true! Or am I missing something?

Reply:
But we could remove `ScreenLockUtil.applyScreenLock(this);` from `onActivityResult()`. The screen is currently white anyway and in `onResume()`  `applyScreenLock()` will be called.